openedx/core/djangoapps/dark_lang/middleware.py

```diff
@@ -91,16 +92,31 @@ def process_request(self, request):
             return
 
         self._clean_accept_headers(request)
-        self._set_site_or_microsite_language(request)
-        self._activate_preview_language(request)
 
-    def _set_site_or_microsite_language(self, request):
+    def process_response(self, request, response):
+        """
+        Prevent user from requesting un-released languages except by using the preview-lang query string.
```

Comment:
```suggestion
        Apply user's dark lang preference as a cookie for future requests.
```
Apparently this used to be activated by a query string, but currently dark lang preference is set using an `/update_lang/` page instead.